sql/core/src/test/resources/sql-tests/inputs/datetime.sql

```diff
@@ -160,3 +156,83 @@ select from_json('{"time":"26/October/2015"}', 'time Timestamp', map('timestampF
+select date_format(timestamp '1986-05-23', null);
+select date_format(timestamp '1986-05-23', 'invalid');
+select date_format(timestamp '1986-05-23', 'yyyy-MM-dd');
+
+select from_unixtime(null);
+select from_unixtime(null , null);
+select from_unixtime(12345 , null);
+select from_unixtime(null , 'invalid');
+select from_unixtime(null , 'yyyy-MM-dd');
+select from_unixtime(12345 , 'yyyy-MM-dd');
+
+select unix_timestamp() = unix_timestamp();
+select unix_timestamp(null);
+select unix_timestamp(null, null);
+select unix_timestamp(null, 'yyyy-MM-dd');
+select unix_timestamp(null, 'invalid');
+select unix_timestamp(cast(null as date), 'yyyy-MM-dd');
+select unix_timestamp(date '1986-05-23');
+select unix_timestamp(date '1986-05-23', null);
+select unix_timestamp(date '1986-05-23', 'invalid');
+select unix_timestamp(date '1986-05-23', 'yyyy-MM-dd');
+select unix_timestamp(cast(null as string), 'yyyy-MM-dd');
+select unix_timestamp('1986-05-23');
+select unix_timestamp('1986-05-23', null);
+select unix_timestamp('1986-05-23', 'invalid');
+select unix_timestamp('1986-05-23', 'yyyy-MM-dd');
+select unix_timestamp(cast(null as timestamp ), 'yyyy-MM-dd');
+select unix_timestamp(timestamp '1986-05-23');
+select unix_timestamp(timestamp '1986-05-23', null);
+select unix_timestamp(timestamp '1986-05-23', 'invalid');
+select unix_timestamp(timestamp '1986-05-23', 'yyyy-MM-dd');
+
+select to_unix_timestamp(null);
+select to_unix_timestamp(null, null);
+select to_unix_timestamp(null, 'yyyy-MM-dd');
+select to_unix_timestamp(null, 'invalid');
+select to_unix_timestamp(cast(null as date), 'yyyy-MM-dd');
+select to_unix_timestamp(date '1986-05-23');
+select to_unix_timestamp(date '1986-05-23', null);
+select to_unix_timestamp(date '1986-05-23', 'invalid');
+select to_unix_timestamp(date '1986-05-23', 'yyyy-MM-dd');
+select to_unix_timestamp(cast(null as string), 'yyyy-MM-dd');
+select to_unix_timestamp('1986-05-23');
+select to_unix_timestamp('1986-05-23', null);
+select to_unix_timestamp('1986-05-23', 'invalid');
+select to_unix_timestamp('1986-05-23', 'yyyy-MM-dd');
+select to_unix_timestamp(cast(null as timestamp ), 'yyyy-MM-dd');
+select to_unix_timestamp(timestamp '1986-05-23');
+select to_unix_timestamp(timestamp '1986-05-23', null);
+select to_unix_timestamp(timestamp '1986-05-23', 'invalid');
+select to_unix_timestamp(timestamp '1986-05-23', 'yyyy-MM-dd');
+
+select to_timestamp(null);
+select to_timestamp(cast(null as string), 'yyyy-MM-dd');
+select to_timestamp(cast(null as string), 'invalid');
+select to_timestamp('1986-05-23');
+select to_timestamp('1986-05-23', null);
+select to_timestamp('1986-05-23', 'invalid');
+select to_timestamp('1986-05-23', 'yyyy-MM-dd');
+
+select to_date(null);
+select to_date(cast(null as string), 'yyyy-MM-dd');
+select to_date(cast(null as string), 'invalid');
+select to_date('1986-05-23');
+select to_date('1986-05-23', null);
+select to_date('1986-05-23', 'invalid');
+select to_date('1986-05-23', 'yyyy-MM-dd');
```

Comment:
the test coverage isn't good, at least we didn't catch https://github.com/apache/spark/pull/28650/files#r431878950

Can we create a table in this test file and test non-foldable format string as well?

Reply:
add more tests to check codegen version